### DISCUSSION TITLE
Can I know If user is private account when searching?

### DISCUSSION THREAD

**Discussion by emreond** [sbertix/Swiftagram#173] [Q&A] [answered]
Hello,

I am searching users with below endpoint. I want to know If returned users have private accounts or followed by me. On previous framework, It was returning friendShip statuses but I guess in this one It returns nil. Is it possible to know that value because I don't wanna call `Endpoint.Friendship.summary` after each search. 

` Endpoint.User.all(matching: string)`

**Comment by emreond** [comment]
I could get values with below code. As I saw, It is returning but val is not in the model.
```       
        let friendShipDictionary = user.wrapper().friendshipStatus.dictionary()
        let isPrivate = friendShipDictionary?["isPrivate"]?.bool() ?? false
        let isRestricted = friendShipDictionary?["isRestricted"]?.bool() ?? false
```

**Comment by sbertix** [comment, marked as answer]
Hey @emreond,

The property you're looking for is `user.access`, an enum having `.default`, `.private` and `.verified` cases.

About the friendship status in `User` models, I simply removed it cause most of the times, as you would know since you were using **SwiftyInsta**, it is not populated and people would complain about it, and I would find myself wasting so much time in explaining them that it's Instagram's "fault" in not returning those anymore.
I'm considering adding it back for `5.0.0` but I'm not 100% convinced, yet.

**Reply by emreond** [reply]
Thank you for your reply. Actually I wanted to know If I could reach that users media/followers etc... so I wanted to get If the user has public profile or private but I am following already. I guess I'll combine `friendShip` with `user.access`

**Reply by sbertix** [reply]
Sorry, I genuinely don't get what you mean in your reply.  
I feel like my answer covers both the "how to see if a user has a private account" and the "why friendship is not there" (as you've mentioned yourself the "workaround"), so I don't really feel how you could have different expectations given the question… 

"Can I know if a user is private when searching?" Again: yes, just check it's `access` property. 
"Can I get the friendship status of a user when searching?" For most `User.Collection` `Endpoint`s no you can't cause Instagram does not populate it anymore. If it is populated it's either in `user["friendshipStatus"]` or `user["friendship"]`, like you said.

Please, if you meant something different in your question that didn't come though, feel free to open a new **Discussions** 😊

**Reply by emreond** [reply]
Actually I wanted to get permission of If I can reach users posts or not. For that, user may have public profile or I am already following him/her. I wanted to get this value in search state but I guess It is impossible.
